src/StringEncodings.jl

```diff
@@ -318,7 +318,10 @@ function fill_buffer!(s::StringDecoder)
         return i
     end
 
-    s.inbytesleft[] += readbytes!(s.stream, view(s.inbuf, Int(s.inbytesleft[]+1):BUFSIZE))
+    # TODO: use a SubArray once readbytes! performance is fixed (JuliaLang/julia#36607):
```

Comment:
Let's wait for https://github.com/JuliaLang/julia/pull/36607 to be merged and add a `VERSION` check.